demos/projects/PC/linux/tests/aux_needed_functions.c

```diff
@@ -0,0 +1,256 @@
+{
+    va_list arg;
+
+    va_start( arg, pcFormat );
+    vprintf( pcFormat, arg );
+    va_end( arg );
+}
+
+/* Needed for compilation */
+extern uint32_t ulApplicationGetNextSequenceNumber( uint32_t ulSourceAddress,
+                                                    uint16_t usSourcePort,
+                                                    uint32_t ulDestinationAddress,
+                                                    uint16_t usDestinationPort )
+{
+    ( void ) ulSourceAddress;
+    ( void ) usSourcePort;
+    ( void ) ulDestinationAddress;
+    ( void ) usDestinationPort;
+
+    return ( uint32_t ) configRAND32();
+}
+/*-----------------------------------------------------------*/
+
+/* Needed for compilation */
+BaseType_t xApplicationGetRandomNumber( uint32_t * pulNumber )
+{
+    *pulNumber = ( uint32_t ) configRAND32();
+    return pdTRUE;
+}
+/*-----------------------------------------------------------*/
+
+/* Needed for compilation */
+void vApplicationGetIdleTaskMemory( StaticTask_t ** ppxIdleTaskTCBBuffer,
+                                    StackType_t ** ppxIdleTaskStackBuffer,
+                                    uint32_t * pulIdleTaskStackSize )
+{
+    ( void ) ppxIdleTaskStackBuffer;
+    ( void ) ppxIdleTaskStackBuffer;
+    ( void ) pulIdleTaskStackSize;
+}
+/*-----------------------------------------------------------*/
+
+static void prvSRand( UBaseType_t ulSeed )
+{
+    /* Utility function to seed the pseudo random number generator. */
+    ulNextRand = ulSeed;
+}
+/*-----------------------------------------------------------*/
+
+UBaseType_t uxRand( void )
+{
+    const uint32_t ulMultiplier = 0x015a4e35UL, ulIncrement = 1UL;
+
+    /* Utility function to generate a pseudo random number. */
+
+    ulNextRand = ( ulMultiplier * ulNextRand ) + ulIncrement;
+    return( ( int ) ( ulNextRand >> 16UL ) & 0x7fffUL );
+}
+/*-----------------------------------------------------------*/
+
+/* Needed for compilation */
+void vApplicationGetTimerTaskMemory( StaticTask_t ** ppxTimerTaskTCBBuffer,
+                                     StackType_t ** ppxTimerTaskStackBuffer,
+                                     uint32_t * pulTimerTaskStackSize )
+{
+    ( void ) ppxTimerTaskStackBuffer;
+    ( void ) ppxTimerTaskStackBuffer;
+    ( void ) pulTimerTaskStackSize;
+}
+/*-----------------------------------------------------------*/
+
+uint64_t ullGetUnixTime( void )
+{
+    return ( uint64_t ) time( NULL );
+}
+/*-----------------------------------------------------------*/
+
+/* Needed for compilation */
+int mbedtls_platform_entropy_poll( void * data,
+                                   unsigned char * output,
+                                   size_t len,
+                                   size_t * olen )
+{
+    FILE * file;
+    size_t read_len;
+
+    ( ( void ) data );
+
+    *olen = 0;
+
+    file = fopen( "/dev/urandom", "rb" );
+
+    if( file == NULL )
+    {
+        return( MBEDTLS_ERR_ENTROPY_SOURCE_FAILED );
```

Comment:
nit - (assuming we keep this logic in 1st place, see my mbedtls_platform_entropy_poll comments above) recommend appropriate logging (along with `errno`) if this is hit, along with any sort of issues with `fread` failing below.